src/vs/platform/storage/common/storage.ts

```diff
@@ -607,13 +617,13 @@ export abstract class AbstractStorageService extends Disposable implements IStor
 
 					const newValue = newStorage.get(key);
 					if (newValue !== oldValue) {
-						this.emitDidChangeValue(scope, key);
+						this.emitDidChangeValue(scope, { key, wasChangedExternally: true });
```

Comment:
What about `newStorage.set(key, value);` in line 608, shouldnt that also `wasChangedExternally: true`?